### DISCUSSION TITLE
Disable outputEscape for some specific filters

### DISCUSSION THREAD

**Discussion by ybouane** [harttle/liquidjs#566] [General]
Hello,
I'm using the outputEscape option to escape output by default.
However, for some specific filters I want to disable the escaping, how can that be done?
How to integrate the functionality of the raw filter into my own filters. (I saw in the code that the raw filter is hardcoded :/ )

Example:
`{{ "<b>Hello \nworld.</b>" }}`
would output:
`&lt;b&gt;hello
world&lt;/b&gt;`

I made a nl2br filter:
`const nl2br = function(str) {return escapeHtml(str).replace(/\r?\n/g, '<br>')};`

So for:
`{{ "<b>Hello \nworld.</b>" | nl2br }}`
the output is:
`&amp;lt;b&amp;gt;hello &lt;br&gt;world&amp;lt;/b&amp;gt;`
(it escapes twice)

Whereas I actually want:
`&lt;b&gt;hello <br>world&lt;/b&gt;`

The way to achieve that would be to type in `{{ "<b>Hello \nworld.</b>" | nl2br | raw }}` (and remove the escapeHtml step from the function), but I wanted to check if it was possible to avoid that?

I tried storing a variable isEscaped in the context of the nl2br filter and checking if is set on my outputEscape function which works in a simple use-case but fails in this example:
`{% assign "a" | nl2br %}
{{ "<b>Hello \nworld.</b>" }}`

In this case, the second statement skips the escaping which is undesirable.

Thanks for your help!

**Comment by harttle** [comment]
[outputEscape](https://liquidjs.com/api/interfaces/liquid_options_.liquidoptions.html#Optional-outputEscape) is `undefined` by default, `{{ "<b>Hello \nworld.</b>" }}` will output `"<b>Hello \nworld.</b>"` directly, by default.

> I tried storing a variable isEscaped in the context of the nl2br filter

You're missing `=`, the correct syntax is:

```liquidjs
{% assign foo = "<a>" | escape %}
```

Live demo: https://liquidjs.com/playground.html#eyUgYXNzaWduIGZvbyA9ICI8YT4iIHwgZXNjYXBlICV9Cnt7IGZvbyB9fQ==,e30=

**Reply by ybouane** [reply]
But I do want outputEscape to be enabled. I just want to disable it for specific filters such as the nl2br filter to avoid having to always use | raw everytine i need to use nl2br.

**Comment by ybouane** [comment]
Is there any solution to this?
Once again, I want outputEscape to be enabled but dynamically disable it if some specific filters are being used.

{{ foo }} // outputEscape is enabled
{{ foo | nl2br }} // outputEscape is disabled

That is because some filters like nl2br already escape the input.

For now the only solution is to do:
{{ foo | nl2br | raw }}

A way to solve this issue would be if it is possible for a filter to know which filters were called before itself?
Or to be able to mark multiple filters as "raw":
https://github.com/harttle/liquidjs/blob/1380ac931a51dcb236c913aba64acc613f81c1ad/src/template/output.ts#L16

**Comment by harttle** [comment]
Hi @ybouane , I took this approach

> Or to be able to mark multiple filters as "raw":

And released this feature in v10.3.0, here's an example:

https://github.com/harttle/liquidjs/blob/e6db371519f0fb3b0068347cfb2016aed386c8fa/test/integration/liquid/register-filters.ts#L53-L62

**Reply by ybouane** [reply]
That's amazing! Thank you so much for addressing this!!
